File: SW/D3_10059.py
std = ["NA", "MMYY", "YYMM", "AMBIGUOUS"]
for t in range(1, int(input())+1):
    N = list(input())
    a, b = int("".join(N[:2])), int("".join(N[2:]))

    res = 0
    #MMYY
    if 1 < a < 13:
        res += 1
    
    #YYMM
    if 1 < b < 13:
        res += 2
    
    print("#{} {}".format(t, std[res]))


    
    

File: SW/D3_10059.java
import java.util.*;

public class D3_10059 {
    static int a, b, c;
    static String[] nums = {"NA", "MMYY", "YYMM", "AMBIGUOUS"};
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        sc.nextLine();
        
        for (int t=1; t<=T; t++) {
            int s = Integer.parseInt(sc.nextLine());
            a = s/100;
            b = s%100;
            
            c = 0;
            if (a > 0 && a < 13) {
                c += 1;
            }

            if (b > 0 && b < 13) {
                c += 2;
            }
            
            System.out.printf("#%d %s\n", t, nums[c]);
        } 
    }    
}
